gcsfs/core.py

```diff
@@ -589,11 +589,17 @@ async def _get_object(self, path):
             # listing.
             raise FileNotFoundError(path)
 
-        result = self._process_object(
-            bucket, await self._call("GET", "b/{}/o/{}", bucket, key, json_out=True)
-        )
-
-        return result
+        res = None
+        # Using a list with a prefix only requires storage.objects.list permissions, rather than
+        # storage.objects.get.
```

Comment:
I've actually come across the reverse too, where you have get permission, but not list permission

Reply:
I wondered about that. What do you think about:
- try get (code in `master`)
- catch the OSError (which is only raised for Forbidden)
- fallback to this prefix method
- if both fail, make sure to return the right error (`Forbidden` if got that originally, otherwise `FileNotFound`)
?

Reply:
Yes, that sounds reasonable. I don't know if it's testable :|